modules/task/task.go

```diff
@@ -52,36 +57,83 @@ func globalBackendStatsClient() *statsClient {
 }
 
 // New creates an async task queue ModuleProvider.
-func New(createFunc BackendCreateFunc) service.ModuleProvider {
+func New(createFunc BackendCreateFunc, options ...ModuleOption) service.ModuleProvider {
 	return service.ModuleProviderFromFunc("task", func(host service.Host) (service.Module, error) {
-		return newAsyncModuleSingleton(host, createFunc)
+		return newAsyncModuleSingleton(host, createFunc, options...)
 	})
 }
 
 func newAsyncModuleSingleton(
 	host service.Host,
 	createFunc BackendCreateFunc,
+	options ...ModuleOption,
 ) (service.Module, error) {
 	_once.Do(func() {
-		_asyncMod, _asyncModErr = newAsyncModule(host, createFunc)
+		_asyncMod, _asyncModErr = newAsyncModule(host, createFunc, options...)
 	})
 	return _asyncMod, _asyncModErr
 }
 
 func newAsyncModule(
 	host service.Host,
 	createFunc BackendCreateFunc,
+	options ...ModuleOption,
 ) (service.Module, error) {
-	backend, err := createFunc(host)
+	config := &Config{}
+	for _, option := range options {
+		if err := option(config); err != nil {
+			return nil, err
+		}
+	}
+	b, err := createFunc(host)
 	if err != nil {
 		return nil, err
 	}
+	mBackend := &managedBackend{b, *config}
 	_globalBackendMu.Lock()
-	_globalBackend = backend
+	_globalBackend = mBackend
 	_globalBackendStatsClient = newStatsClient(host.Metrics())
 	_globalBackendMu.Unlock()
-	return backend, nil
+	return mBackend, nil
+}
+
+// managedBackend is the root for all backends and controls execution
+type managedBackend struct {
+	Backend
+	config Config
+}
+
+// Start implements the Module interface
+func (b *managedBackend) Start() error {
+	if err := b.Backend.Start(); err != nil {
+		return errors.Wrap(err, "unable to start backend")
+	}
+	if !b.config.DisableExecution {
+		errorCh := b.ExecuteAsync()
```

Comment:
If someone else gets b.ExecuteAsync() and will read errors, does this mean that they are going to lose some errors?